File: Release/WetiMatlabFunctions/PostProcessing/CollectTimeResults.m
function TimeResults = CollectTimeResults(DataFiles,PostProcessingConfig,options)

% inputs
arguments
    DataFiles               cell
    PostProcessingConfig    struct
    options.nCore           {mustBeInteger} = maxNumCompThreads % set option to 0 for no parallel processing
end

% get dimensions
nDataFiles  = size(DataFiles,1);
TimeResults = cell(nDataFiles,1);

% loop over data files
parfor (iDataFile = 1:nDataFiles,options.nCore) 
    ThisDataFile    = DataFiles{iDataFile};
    [~,~,EXT]       = fileparts(ThisDataFile);
    TSC             = []; % to avoid warning of "Uninitialized Temporaries"
    switch EXT
        case '.outb'
            TSC     = ReadFASTbinaryIntoTSC(ThisDataFile);    
        case '.res'
            TSC     = ReadFlex5IntoTSC(ThisDataFile);
        case '.mat'            
            Dummy   = load(ThisDataFile,'TSC');
            TSC     = Dummy.TSC
        otherwise
            error('Currently %s files are not supported!',EXT)
    end
    % keep only some channels, if requested
    if isfield(PostProcessingConfig,'DataChannels')&&~isempty(PostProcessingConfig.DataChannels)
        Channels  	= PostProcessingConfig.DataChannels;
        TsNames     = gettimeseriesnames(TSC);
        RemoveIdx   = find(~ismember(TsNames,Channels));
        for iTsName = RemoveIdx
            TSC     = removets(TSC,TsNames(iTsName));
        end 
    end
    % load into cell
    TimeResults{iDataFile} = TSC;
end

end

File: Release/PythonFunctions/PostProcessing/CollectTimeResults.py
from joblib import Parallel, delayed
import os
import scipy.io as sio
from FileOperations.ReadFASTbinaryIntoTSC import ReadFASTbinaryIntoTSC
def CollectTimeResults(data_files, post_processing_config, n_core=None):
    if n_core is None:
        n_core = os.cpu_count()

    n_data_files = len(data_files)
    time_results = [None] * n_data_files

    # loop over data files
    for i_data_file in range(n_data_files):
        this_data_file = data_files[i_data_file]
        _, ext = os.path.splitext(this_data_file)
        tsc = None  # to avoid warning of "Uninitialized Temporaries"
        ext = '.outb'
        if ext == '.outb':
            tsc = ReadFASTbinaryIntoTSC(this_data_file)
        elif ext == '.res':
            tsc = ReadFlex5IntoTSC(this_data_file) #this function is not written yet
        elif ext == '.mat':
            dummy = sio.loadmat(this_data_file)
            tsc = dummy['TSC']
        else:
            raise ValueError(f'Currently {ext} files are not supported!')

        # keep only some channels, if requested
        if 'DataChannels' in post_processing_config and post_processing_config['DataChannels']:
            channels = post_processing_config['DataChannels']
            ts_names = list(TSC.columns)
            remove_idx = [i for i, name in enumerate(ts_names) if name not in channels]
            for i_ts_name in remove_idx:
                tsc = remove_ts(tsc, ts_names[i_ts_name])

        # load into cell
        time_results[i_data_file] = tsc

    return time_results
